shared/js/ui/models/whitelist.es6.js

```diff
@@ -44,9 +50,27 @@ Whitelist.prototype = window.$.extend({},
           value: true
         }
         })
+
+        this.setWhitelistFromSettings()
       }
 
       return domain
+    },
+
+    setWhitelistFromSettings: function () {
+      let self = this
+      this.fetch({getSetting: {name: 'whitelisted'}}).then((list) => {
+        list = list || {}
+        let wlist = Object.keys(list)
+        wlist.sort()
+
+        // Prevent sealing array elements
+        self.list = [...wlist]
```

Comment:
I'm not sure what the benefit of `[...wlist]` is. `wlist` is an array of strings already, why unpack and wrap it again?

Reply:
Because with regular assignment somehow self.list gets sealed and it throws an error when I try to splice() on it "TypeError: can't add/remove sealed array elements".
Unpacking and wrapping again seems like the only way around this.

Reply:
Ohhh OK, that makes sense. I guess anything passed via `fecth` is sealed and so we can't add changes? We don't seem to be doing it on our side, at least.

Can you explain more in the comment?

Reply:
Yeah added a longer comment, hope it's helpful